### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by calebchris000**
# Highlights 🌟 

- First off, the ``current_user`` has been removed, because devise has a functional method.
- The authentication, which is awesome, is up and running now.
- Signup, reset, login are all working.

**Peace.** ✌️

### CODE REVIEW

**Review by HAMMAS-SALEEM — COMMENTED** [review]
# Invalid Submission :red_circle:

invalid-submission

## Hey @calebchris000 :wave:

Good job on implementing some of the requirements but I cannot proceed review of your pull request because there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

# Mandatory Changes ♻️

- [ ] It would be nice if you remove your Live Demo section completely from your readme file to make your readme descriptive and professional 👍
- [ ] It would be nice if your partner have commits in this pull request since it is a pair-programming activity 👍

**Review by Gambit142 — COMMENTED** [review]
Hi @calebchris000 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- This is a pair-programming activity. You were the only one that made contributions to the success of this milestone I would advise you to close this pull request and create another one where you and your partner contributed to the success. 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/author.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class Author < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (GzimAsani) [inline app/models/author.rb:5]: - [ ] The application need to ask for confirmation of email. So please add `:confirmable` to your devise as well

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi #calebchris000, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
### Hello, @calebchris000                         :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with implementing all the devise features except confirmable, it looks great +1: 

 ## Required Changes 

- [ ] Check for the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by calebchris000** [comment]
Hi **Code Reviewer**, @GzimAsani ,

This PR got approved, but the button didn't appear to 'Submit'. I would appreciate it if this is rectified.
Thanks.

**Review by franclobo — COMMENTED** [review]
Hi @calebchris000, 

descarga

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has an incorrect Git flow, please 🙏🏽 use fresh feature branches for each milestone. You have already approved the **add devise** milestone now you have to create a new branch to push the changes to approve **Blog app - integration specs for views and fix n+1 problems** activity.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by GzimAsani** [comment]
> Hi **Code Reviewer**, @GzimAsani ,
> 
> This PR got approved, but the button didn't appear to 'Submit'. I would appreciate it if this is rectified. Thanks.

I did write approved by mistake, you need to make the required changes based on the comment that I left. Also for all milestones you need to create new pull request. I guess you requested both milestones on the same time that is why I checked otherwise it should've been invalidated

**Comment by nicupop729** [comment]
@c00p75  please note that this is a pair-programming activity where the students need to ask for review individually with their own PR.

The approval given by @bobb-Rob will be reverted in the dashboard for @c00p75 .

In order to proceed further to the next milestone, @c00p75 , kindly ask for another review using a PR from your own repository. Please don't forget to collaborate in this activity with your pair-programming partner.

We apologize for any inconvenience.

Good luck!

**Pull request merged**